### PR TITLE
Feature/eth watcher

### PR DESCRIPTION
**Pull request by tkmct**
Add event watcher

### CODE REVIEW

File: event-watcher/src/event_watcher.rs [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+    stream: EventFetcher<T>,
+    listeners: Vec<Box<dyn Fn(&Log) -> () + Send>>,
+}
+
+impl<T> EventWatcher<T>
+where
+    T: EventDB,
+{
+    pub fn create(url: &str, address: Address, abi: Vec<Event>, db: T) -> EventWatcher<T> {
+        let (_eloop, transport) = web3::transports::Http::new(url).unwrap();
```
Comment (syuhei176) [inline event-watcher/src/event_watcher.rs]: Maybe _eloop will be stopped by dropping.
https://tomusdrw.github.io/rust-web3/web3/transports/struct.EventLoopHandle.html

**Comment by tkmct** [comment]
@syuhei176 I could solve the problem. Thank you!

File: Cargo.toml [hunk 2]
```diff
@@ -19,6 +19,7 @@ members = [
 	"db",
 	"operator",
 	"predicate-plugins",
+  "event-watcher",
```
Comment (syuhei176) [inline Cargo.toml]: different indent?

File: event-watcher/src/event_watcher.rs [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+        let stream = EventFetcher::new(web3, address, abi, db);
+
+        EventWatcher {
+            _eloop: eloop,
+            stream,
+            listeners: vec![],
+        }
+    }
+
+    pub fn subscribe(&mut self, listener: Box<dyn Fn(&Log) -> () + Send>) {
```
Comment (syuhei176) [inline event-watcher/src/event_watcher.rs]: Also we need unsubscribe, it's ok in another PR.

File: event-watcher/src/event_db.rs [hunk 4]
```diff
@@ -0,0 +1,69 @@
+use ethabi::Hash;
+use std::collections::HashMap;
+
+pub trait EventDB {
```
Comment (syuhei176) [inline event-watcher/src/event_db.rs]: EventDB should be merged to db crate. in next PR.

Comment (syuhei176) [inline event-watcher/src/event_db.rs]: EventDB is better to be EventDb in Rust.

Comment (tkmct) [reply event-watcher/src/event_db.rs]: https://rust-lang-nursery.github.io/api-guidelines/naming.html
I found it here. Thanks!

**Comment by tkmct** [comment]
@syuhei176 I fixed issues. Please take a look.

**Comment by tkmct** [comment]
I put it in single commit.

**Pull request merged**